packages/playwright-test/src/plugins/vitePlugin.ts

```diff
@@ -92,7 +92,8 @@ export function createPlugin(
       // 3. Check component sources.
       const sourcesDirty = !buildExists || hasNewComponents || await checkSources(buildInfo);
 
-      viteConfig.root = rootDir;
+      viteConfig.root = templateDir;
```

Comment:
At least when I started with it (Vite 2), it did not work - Vite seemed to only process / transpile files under the root. In this case, component files become outside of the /playwright/ root, so I wonder if things will keep working by the spec...

Reply:
If things are indeed working with this new setup, let's leave old behavior behind environment variable, so that it was easy for the existing users to keep it working without intrusive changes.

Reply:
> At least when I started with it (Vite 2), it did not work - Vite seemed to only process / transpile files under the root. In this case, component files become outside of the /playwright/ root, so I wonder if things will keep working by the spec...

Will test it further but ATM i think there is one thing left todo:
Vite doesn't pick up the config files that are in the root of the project. So config files like `svelte.config.js` and `tailwind.config.js` have to be duplicated and placed inside the `/playwright/` as well.

Reply:
> If things are indeed working with this new setup, let's leave old behavior behind environment variable, so that it was easy for the existing users to keep it working without intrusive changes.

Curious if you ever thought about a post-install script that migrates old API's to the new ones? As in this case that would be finding and replacing the import path's

Reply:
Found a way without breaking existing users by moving `index.html` one directory down after build